### ISSUE TITLE
'Missing input file' when I build tensorflow-upstream

### ISSUE THREAD

**Issue by ws7474741** [ROCmSoftwarePlatform/tensorflow-upstream#855]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version:  1.14.0
- Python version: 3.5.2
- Bazel version (if compiling from source):  1.1.0
- GCC/Compiler version (if compiling from source):  5.4.0
- ROCm/MIOpen version: ROCm 3.0.6 / MIOpen 2.2.0.7617-rocm-rel-3.0-6-948938d
- GPU model and memory: AMD Vega 64

**Describe the problem**
ERROR: missing input file '@local_config_rocm//rocm:rocm/include/hipcub/hipcub_version.hpp'
ERROR: missing input file '@local_config_rocm//rocm:rocm/include/rocprim/rocprim_version.hpp'
ERROR: /home/rocm-user/.cache/bazel/_bazel_rocm-user/3181dc364f7e690f6fd1939d3f4c52e5/external/local_config_rocm/rocm/BUILD:124:1: @local_config_rocm//rocm:rocprim: missing input file '@local_config_rocm//rocm:rocm/include/hipcub/hipcub_version.hpp'
ERROR: /home/rocm-user/.cache/bazel/_bazel_rocm-user/3181dc364f7e690f6fd1939d3f4c52e5/external/local_config_rocm/rocm/BUILD:124:1: @local_config_rocm//rocm:rocprim: missing input file '@local_config_rocm//rocm:rocm/include/rocprim/rocprim_version.hpp'
Target //tensorflow/tools/pip_package:build_pip_package failed to build
ERROR: /home/rocm-user/tensorflow-upstream/tensorflow/python/tools/BUILD:224:1 2 input file(s) do not exist
INFO: Elapsed time: 29.412s, Critical Path: 10.09s
INFO: 200 processes: 200 local.
FAILED: Build did NOT complete successfully

**Provide the exact sequence of commands / steps that you executed before running into the problem**

I could find above two files in *"/opt/rocm/include/hipcub/hipcub_version.hpp"* and *"/opt/rocm/include/rocprim/rocprim_version.hpp"*.

But when I build tensorflow-upstream by using:
`$ ./build_rocm_python3` to build
It shows above problem.

**Comment by sunway513** [comment]
Hi @ws7474741 , most likely the issue is related to your local environment. 
The following docker container has full dependencies to build TF1.14 code base from source with ROCm3.0 user bits:
**rocm/tensorflow:rocm3.0-tf1.14-dev**

Can you try to check there and do a side-by-side with your local environment? For steps to deploy docker containers, please refer to the following document:
https://hub.docker.com/r/rocm/tensorflow

**Comment by ws7474741** [comment]
Hi @sunway513 , it works! Thank you so much!

**Comment by ianzur** [comment]
I ran into a similar issue when building `r2.7-rocm-enhanced` after I upgraded rocm.

Before the initial build attempt I hadn't installed hipcub. After installing this dependency bazel reported: (even after a `bazel clean`)
```
ERROR: /home/ian/Documents/projects/tensorflow-upstream/tensorflow/python/BUILD:398:23: //tensorflow/python:py_func_headers_lib: missing input file 'external/local_config_rocm/rocm/rocm/include/hipcub/hipcub_version.hpp', owner: '@local_config_rocm//rocm:rocm/include/hipcub/hipcub_version.hpp'
```

Fix: setting expunge to true actually cleaned everything.
[`bazel clean --expunge`](https://docs.bazel.build/versions/main/command-line-reference.html#flag--expunge)


<details><summary>sysinfo</summary>

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian 11 Bullseye x86_64 Linux 5.10.0-11-amd64
- Python version: 3.9.2
- Bazel version (if compiling from source): 3.7.2
- Tensorflow version: v2.7.0-3120-gbc1a398b53a
- GCC/Compiler version (if compiling from source): 10.2.1
- ROCm/MIOpen version: 4.5.2
- GPU model and memory: gfx900, Radeon Vega Frontier Edition, 16GB

</details>
